tools/assetpipelineeditor/camerawidget.cpp

```diff
@@ -227,6 +256,61 @@ void CameraWidget::setFoV(double fov)
         m_camera->setFieldOfView(fov);
 }
 
+void CameraWidget::save()
+{
+    QSettings settings;
+    const QString lastPath = settings.value(LASTPATHSETTING, QDir::homePath()).toString();
+    const auto filename = QFileDialog::getSaveFileName(
+                this,
+                QLatin1String("Save Camera Settings"),
+                lastPath,
+                QLatin1String("*.apecam"));
+
+    if (filename.isEmpty())
+        return;
+
+    QFileInfo fi(filename);
+    settings.setValue(LASTPATHSETTING, fi.absolutePath());
+
+    QFile saveFile(filename);
+    if (!saveFile.open(QIODevice::WriteOnly)) {
+        qWarning("Couldn't open camera settings save file.");
+        return;
+    }
+
+    QJsonObject cameraObject;
+    write(cameraObject);
```

Comment:
write/read are a bit confusing as they are usually related to either the system calls or QIODevice read/write. How about doing QJsonObject cameraObject = toJson(); and fromJson() to restore?